app/processing/schedule/get-scheduled-settlements.js

```diff
@@ -0,0 +1,24 @@
+const moment = require('moment')
+const db = require('../../data')
+const config = require('../../config').processingConfig
+
+const getScheduledSettlements = async (started, transaction) => {
+  return db.schedule.findAll({
```

Comment:
Aaahh gotcha, gotcha

Does this need some sorta 'limit to 1st x amount of rows' w/in the `findAll` if sucha param exists?

Just some thoughts on the replica sitch:
Does replica A pick up all schedules (say idk 2k+) which causes its memory/CPU to spike so K8s brings up another replica, so replica B looks to pick up all in that table but at this point (just given the domain of this) is there likely to be any/many new records tht won't have been picked up by replica A? I guess this is v sensitive to our configured Helm values

Reply:
Yes good point.  You can set that with `limit: 100` for example then it will only lock 100 rows.  Then next replica will skip those locked rows and pick up the next 100.

You're correct, it depends on what we setup in Helm on what the behaviour will be.  So I'll describe a few of those options.

The actual value needs some thought (and testing) as we want to maximise throughput and the answer is subjective to each service.  For example is it more performant to keep WIP low for each replica but have a greater number of replicas?  Or is it better to have less replicas with greater WIP.  Greater WIP may mean those pods to have more CPU/Memory allocated to them.

### Set number of `replicas` > 1
In the helm chart you can define how many instances of the service should be running independent of autoscaling.  So if we set that value to `4` there will always be four pods running ready to process. 

### Use horizontal pod autoscaling
We declare a threshold of usage for CPU and/or memory usage of a pod and if that capacity is exceeded then another replica is created up to the maximum number of replicas we permit in the helm chart.

That threshold can either be a fixed CPU/memory usage or a percentage of usage against the limit.  The latter is what we tend to do.

It's a bit of an art to get the balance right of how much memory and CPU we allocate to a pod and at what point we consider it under high enough load to consider scaling.  We'd need to identify what idle usage is vs under strain usage is so we set the appropriate values.

### Use non-standard metrics
With Kubernetes you can use custom metrics to control scaling beyond the default memory and CPU.  There are also tools such as Keda that can help with this.  For example, rather than look at pod utilisation, we could look at how many messages are waiting to be processed on a service bus subscription and use that to scale the service.  If there are only a few messages then just keep replicas low and work through them.  If we get a surge of thousands of messages then start rapidly creating more replicas to handle the load.

We do kind of get that anyway as lots of messages tends to result in CPU/memory spikes, but there are ways to make it more predictable.